api/proto/sign/v1alpha2/sign_service.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sign.v1alpha2;
+
+import "core/v1alpha2/record.proto";
+import "google/protobuf/empty.proto";
+
+service SignService {
+  // Sign agent records using one of the auth methods
+  rpc SignOIDC(SignOIDCRequest) returns (SignOIDCResponse);
+  rpc SignWithKey(SignWithKeyRequest) returns (SignWithKeyResponse);
+
+  // Verify signed agent records using on of the auth methods
+  rpc VerifyOIDC(VerifyOIDCRequest) returns (google.protobuf.Empty);
+  rpc VerifyWithKey(VerifyWithKeyRequest) returns (google.protobuf.Empty);
```

Comment:
It would be nice to add a short description of the two available method: OIDC-based identity signing vs traditional key-based signing.